Modules/_posixsubprocess.c

```diff
@@ -482,6 +495,17 @@ child_exec(char *const exec_array[],
         POSIX_CALL(setsid());
 #endif
 
+    if (call_setgid) {
+#ifdef HAVE_SETGROUPS
+        POSIX_CALL(setgroups((groups_size-1), &groups[1]));
+#endif /* HAVE_SETGROUPS */
+        POSIX_CALL(setregid(groups[0], groups[0]));
```

Comment:
@gpshead Do we need to add configure checks for `setreuid/setregid`? We already have checks for `setuid/setgid` and `seteuid/setegid`.

Reply:
yep we should add them.  (how? make a configure.ac edit to add them to the list if of apis it checks for and rerun autoconf to regenerate the configure script)

Reply:
It looks like we already have these checks in configure.ac
https://github.com/python/cpython/blob/master/configure.ac#L3513

Reply:
You're right, I missed the macros while looking at my `pyconfig.h`. So what remains is to add checks for appropriate macros before using the functions. Also, if a function is missing, I think that an exception should be raised if the corresponding keyword argument is used (you could do the checks on Python side using something like `hasattr(os, 'setreuid')`). That applies to `setgroups` as well: can't silently ignoring supplementary groups result in a security issue?   

But please hold on: after spending some time on the PR, I realized that I poorly understand the motivation behind the particular choice of `setreuid/setregid` functions, as well as how supplementary groups should be handled. See https://bugs.python.org/msg336346.

Reply:
I can add the macro checks to the C code, and make it fail on the python side if the functions aren't available. What exception should I use for this case, RuntimeError?

Reply:
Looking at existing code, such errors are reported as `ValueError`: https://github.com/python/cpython/blob/53b9e1a1c1d86187ad6fbee492b697ef8be74205/Lib/subprocess.py#L731-L745